### PR TITLE
3.5 into main

### PR DESCRIPTION
**Pull request by jack-w-shaw**
Forward merge:
- https://github.com/juju/juju/pull/16846
- https://github.com/juju/juju/pull/16863
- https://github.com/juju/juju/pull/16867
- https://github.com/juju/juju/pull/16878
- https://github.com/juju/juju/pull/16875
- https://github.com/juju/juju/pull/16879

Conflicts:
- agent/agentbootstrap/bootstrap.go
- apiserver/facades/client/modelmanager/modelinfo_test.go
- apiserver/facades/client/modelmanager/modelmanager.go
- apiserver/facades/client/modelmanager/register.go
- apiserver/facades/controller/undertaker/undertaker.go
- apiserver/facades/controller/undertaker/undertaker_test.go
- apiserver/facades/schema.json
- cmd/jujud/agent/agenttest/agent.go
- cmd/jujud/agent/machine/manifolds.go
- database/bootstrap.go
- featuretests/dblog_test.go
- internal/database/bootstrap_test.go
- internal/worker/dbaccessor/manifold.go
- internal/worker/dbaccessor/manifold_test.go
- internal/worker/dbaccessor/package_test.go

Conflicts mostly arose from https://github.com/juju/juju/pull/16863 (@SimonRichardson) and https://github.com/juju/juju/pull/16875 (@ me)

### CODE REVIEW

File: cmd/jujud/agent/machine/manifolds.go [hunk 1]
```diff
@@ -926,6 +928,22 @@ func IAASManifolds(config ManifoldsConfig) dependency.Manifolds {
...
+			Clock:                config.Clock,
+			Hub:                  config.CentralHub,
+			Logger:               loggo.GetLogger("juju.worker.dbaccessor"),
+			LogDir:               agentConfig.LogDir(),
+			PrometheusRegisterer: config.PrometheusRegisterer,
+			NewApp:               dbaccessor.NewApp,
+			NewDBWorker:          dbaccessor.NewTrackedDBWorker,
+			NewMetricsCollector:  dbaccessor.NewMetricsCollector,
+			NewNodeManager:       dbaccessor.IAASNodeManager,
+		})),
```
Comment (SimonRichardson) [inline cmd/jujud/agent/machine/manifolds.go:945]: @jack-w-shaw you need to copy this to `CAASManifolds` and ensure the has the following: `NodeNodeManager: dbaccessor.CAASNodeManager`

File: internal/worker/dbaccessor/manifold.go [hunk 2]
```diff
@@ -172,3 +179,15 @@ func dbAccessorOutput(in worker.Worker, out interface{}) error {
...
+
+// IAASNodeManager returns a NodeManager that is configured to use
+// the cloud-local TLS terminated address for Dqlite.
+func IAASNodeManager(cfg agent.Config, logger Logger, slowQueryLogger coredatabase.SlowQueryLogger) NodeManager {
+	return database.NewNodeManager(cfg, false, logger, slowQueryLogger)
+}
+
+// CAASNodeManager returns a NodeManager that is configured to use
+// the loopback address for Dqlite.
+func CAASNodeManager(cfg agent.Config, logger Logger, slowQueryLogger coredatabase.SlowQueryLogger) NodeManager {
```
Comment (SimonRichardson) [inline internal/worker/dbaccessor/manifold.go:191]: This needs wiring up in the CAASManifolds.

**Review by SimonRichardson — CHANGES_REQUESTED** [review]

File: cmd/jujud/agent/machine/manifolds.go (hunk 1 above)
Comment (jack-w-shaw) [reply cmd/jujud/agent/machine/manifolds.go:945]: Done

File: internal/worker/dbaccessor/manifold.go (hunk 2 above)
Comment (jack-w-shaw) [reply internal/worker/dbaccessor/manifold.go:191]: Done

**Review by SimonRichardson — APPROVED** [review]

**Pull request merged**